### DISCUSSION TITLE
Server action redirect from one route group to another

### DISCUSSION THREAD

**Discussion by woutrbe** [vercel/next.js#66541] [Help]
### Summary

I'm running into this incredibly strange issue where I have a login page, with a server action that logs in the user, and then redirects them to `/test`. However there's no redirect happening, in the server logs I can see my `/test` route being rendered. but without a redirect.

My project structure looks like this, this should give me `/test`, `/app/test`, `/auth/login` and `/auth/test`.
```
app
--test
----page.tsx
--(app)
----test
------page.tsx
--(auth)
----auth
------login
--------page.tsx
------test
--------page.tsx
```

I have reduced my server action to simply:
```
export const login = async (prevState: any, formData: FormData): Promise<any> => {
  redirect(`/app/test`, RedirectType.push);
}
```

However when I change the redirect from `/app/test` to `/auth/test`, I'm getting a redirect. (`redirect(`/auth/test`, RedirectType.push);`)

I can navigate to all of the `/test` routes manually in the browser, they all render fine without errors.

I don't fully understand what's happening, but as far as I can deduce, it seems to struggle with redirecting between route groups.

### Additional information

```js
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.4.0: Fri Mar 15 00:12:41 PDT 2024; root:xnu-10063.101.17~1/RELEASE_ARM64_T8103
  Available memory (MB): 16384
  Available CPU cores: 8
Binaries:
  Node: 20.11.0
  npm: 10.2.4
  Yarn: 1.22.22
  pnpm: N/A
Relevant Packages:
  next: 14.2.3 // Latest available version is detected (14.2.3).
  eslint-config-next: 14.2.3
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.4.5
Next.js Config:
  output: N/A
```

**Comment by sebasaenz** [comment]
Same issue here

**Reply by icyJoseph** [reply]
In the network tab, can you see the server action request that gets redirected? Post a screenshot of the headers

**Reply by sebasaenz** [reply]
After the submit it's getting redirected back to `/login` so nothing related to the intended destination (`/home`) is showing in the network tab. The main requests I see coming up after the submit are to `(routes)/layout.js` and to `(emptyLayout)/login/page.js`

**Reply by yota88** [reply]
I'm running into the exact same issue described. Any follow-up on the resolution?

**Comment by Alisadaintanvir** [comment]
same issue I am facing. Any solution u got buddy??

**Comment by RanjithKannan03** [comment]
Same issue here but I was able to do this in a previous project. Suddenly it doesn't work anymore.

**Comment by iankduffy** [comment]
Having the same issue here as well. trying to redirect to another route group with different layout. 

I can see in the dev logs that it has compiled the page and see console logs that exist in the page.tsx.

**Comment by iankduffy** [comment]
Creating a Layout.tsx at the root of app directory (outside of router groups), with base html and body tags seems to have worked for me.

**Reply by Alisadaintanvir** [reply]
have different layout for different route.thats why I created different route group.

**Reply by iankduffy** [reply]
The Layout at the root can be just the HTML tag, allowing you to have different nav, headers and footer: 

```
import { Open_Sans } from "next/font/google";
import "@/styles/global.scss";

const openSans = Open_Sans({ subsets: ["latin"] });

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
      <html lang="en" className={openSans.className}>
        <body>{children}</body>
      </html>
  );
}

```

Then in the router groups you have a layout.tsx, which focusing that router group.

**Reply by Aurelian-C** [reply]
Thanks @iankduffy , your solution works perfectly.
